bin/run

```diff
@@ -1,14 +1,24 @@
-#!/usr/bin/env node
+#!/bin/sh
 
-const fs = require('fs');
-const path = require('path');
-const project = path.join(__dirname, '../tsconfig.json');
-const dev = fs.existsSync(project);
+# Get the local node_modules path;
+# if NODE_PATH is already set, append it to the existing value.
+if [ -n "$NODE_PATH" ]; then
+  UPDATED_PATH="${NODE_PATH}:${PWD}/node_modules"
+else
+  UPDATED_PATH="${PWD}/node_modules"
+fi
 
-if (dev) {
-  require('ts-node').register({ project });
-}
+BASEDIR=$(dirname "$0")
 
-require(`../${dev ? 'src' : 'lib'}`)
-  .SyncDb.run()
-  .catch(require('@oclif/errors/handle'));
+# Run the CLI with NODE_PATH env variable including the path to
+# the local project's node_modules (from the current working directory).
+#
+# Use-case:
+# The main reason for this is to allow the CLI's node process
+# to be able to resolve local project dependencies;
+# this is important when the CLI is installed globally and
+# the local project installation has some of the packages that the CLI needs.
+# For eg: if the local project has mssql package installed and the CLI is invoked globally.
+#
+# This allows the CLI's require calls to fallback to the local dependencies if needed.
+NODE_PATH="${UPDATED_PATH}" node $BASEDIR/run.js
```

Comment:
So basically what we are saying is we don't care about Windows. This doesn't work in windows.

Reply:
Do we need to make this a global CLI tool? I think we should opt out of that because we depend on packages like mssql, psql and knex.

Reply:
Makes sense.

Reply:
Yes, good catch. Need to tweak the `run.cmd` a bit for this to make it work.

Reply:
@mesaugat This is what this PR does. The CLI is not tied with the local dependencies - be it global or local, it's the user's choice. It just supports it both.

When installed globally, it resolves the local dependencies from the local project directory normally. So, there's no dependency in between them. 

Other two reasons are:
 - Right now, it's difficult to test. When you make any change you cannot test it locally unless you publish it as a package and install it in your local project - this is painful and slow. But #50 makes it seamless - you can do `/path/to/local/bin/run` from your project.
- Secondly, you could publish a docker image for sync-db - so the user doesn't even need to install sync-db in their local - it would be like a binary. They'll just need their local deps like mssql & configurations; not even knex is required.

Reply:
@kabirbaidhya I get all of that and I feel this is not required. This is true for any kind of package and we don't need to publish it. We just need do `yarn pack` or `npm pack` and create a package locally. Maybe let the CI do this. Then, we need to install that package from the locally zipped tar to test.

Reply:
> A binary should mostly be self-sufficient. 

`sync-db` always needs a SQL client package to run. So creating a binary that doesn't do anything by itself doesn't make any sense to me. Either we have to package the individual client drivers with the binary or stop thinking about running it globally.

"Someone installs sync-db globally and then wants to run it in their project. Now, they need to install the SQL client and add it to their `package.json` as a local dependency" or maybe install the driver globally as well." 

There are simply too many confusions with the global approach. A simple install these dependencies and run it locally like every other JavaScript package is the best approach in my opinion.

Reply:
@mesaugat I have a different perspective on this. 
> A binary should mostly be self-sufficient.

Actually, the binary is self-dependent. It is just that the project needs to have a vendor specific connector to talk to the corresponding DB they use; but that isn't a part of CLI itself. And, they'll need configuration as per their project's source - that is project specific too. So, actually you can think of the CLI as a self-sufficient binary as one global installation would work with multiple projects configured differently.

But, I also agree to the point that global installation might be confusing for users and might be harder for us to push BC breaking changes in longer run. 

> We just need do yarn pack or npm pack and create a package locally. 

This is not sustainable; testing locally is really painful w/ this approach and this won't work as it's very troublesome at least for sync-db. Because there are various things you'll need to test trying out in different projects before you can push.

So, net-net, we'll need something like this:
1. don't encourage global install -- Yes
2. have this as a separate script (won't be published) to run locally eg: `run.sh`

Thoughts?

Reply:
Cool.